### ISSUE TITLE
[Bug]: `esbonio` fails to start with `ImportError: cannot import name 'WorkspaceConfigurationParams' from 'lsprotocol.types'`

### ISSUE THREAD

**Issue by mion00** [vscode-restructuredtext/vscode-restructuredtext#420]
### Expected behavior

I would like to build and preview the documentation using the Sphinx builder.

### Actual behavior

The `esbonio` server fails to start when opening a `rst` documentation file.

### Detailed steps

I have a Python project with a `docs/` subfolder containing the `.rst` files that are compiled via Sphinx.
After having installed the sphinx libraries and themes I use, I am prompted by the extension to install `esbonio`, which I promptly do.
When the installation completes however I have the message _The Esbonio Language Server server crashed 5 times in the last 3 minutes. The server will not be restarted._

In addition, after restarting VSCode I have again the notification _Esbonio language server is not installed or out of date._, even if I see from the logs that esbonio is indeed installed:

```bash
[Log - 12:14:42 PM] Started to install Esbonio...
[Log - 12:14:42 PM] Running cmd: "/home/mion00/.pyenv/versions/3.10.8-debug/envs/appdaemon/bin/python" -m pip install esbonio --upgrade
[Log - 12:14:48 PM] Successful exec
[Log - 12:14:48 PM] Finished installing esbonio
```
I don't know wether to file an issue with `esbonio` upstream or if this extension is having some unexpected issues..

### Extension information

v189.3.0

### Visual Studio Code information

1.74.3

### Operating system information

Linux Mint 20.3 Cinammon

### Python information

3.10.8

### Sphinx information

6.1.3

### Relevant log output

```shell
[Log - 11:53:44 AM] Running cmd: "/home/mion00/.pyenv/versions/3.10.8-debug/envs/appdaemon/bin/python" -c "import pip; from distutils.version import LooseVersion; print(LooseVersion(pip.__version__) < LooseVersion('20.0.2'))"
[Log - 11:53:45 AM] Successful exec
[Log - 12:01:25 PM] Running cmd: "/home/mion00/.pyenv/versions/3.10.8-debug/envs/appdaemon/bin/python" -c "import esbonio.lsp; from distutils.version import LooseVersion; print(LooseVersion(esbonio.lsp.__version__) < LooseVersion('0.11.2'))"
[Log - 12:01:28 PM] Error
Command failed: "/home/mion00/.pyenv/versions/3.10.8-debug/envs/appdaemon/bin/python" -c "import esbonio.lsp; from distutils.version import LooseVersion; print(LooseVersion(esbonio.lsp.__version__) < LooseVersion('0.11.2'))"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/mion00/.pyenv/versions/3.10.8-debug/envs/appdaemon/lib/python3.10/site-packages/esbonio/lsp/__init__.py", line 52, in <module>
    from .rst import CompletionContext
  File "/home/mion00/.pyenv/versions/3.10.8-debug/envs/appdaemon/lib/python3.10/site-packages/esbonio/lsp/rst/__init__.py", line 37, in <module>
    from pygls.server import LanguageServer
  File "/home/mion00/.pyenv/versions/3.10.8-debug/envs/appdaemon/lib/python3.10/site-packages/pygls/server.py", line 29, in <module>
    from lsprotocol.types import (
ImportError: cannot import name 'WorkspaceConfigurationParams' from 'lsprotocol.types' (/home/mion00/.pyenv/versions/3.10.8-debug/envs/appdaemon/lib/python3.10/site-packages/lsprotocol/types.py)

Error: Command failed: "/home/mion00/.pyenv/versions/3.10.8-debug/envs/appdaemon/bin/python" -c "import esbonio.lsp; from distutils.version import LooseVersion; print(LooseVersion(esbonio.lsp.__version__) < LooseVersion('0.11.2'))"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/mion00/.pyenv/versions/3.10.8-debug/envs/appdaemon/lib/python3.10/site-packages/esbonio/lsp/__init__.py", line 52, in <module>
    from .rst import CompletionContext
  File "/home/mion00/.pyenv/versions/3.10.8-debug/envs/appdaemon/lib/python3.10/site-packages/esbonio/lsp/rst/__init__.py", line 37, in <module>
    from pygls.server import LanguageServer
  File "/home/mion00/.pyenv/versions/3.10.8-debug/envs/appdaemon/lib/python3.10/site-packages/pygls/server.py", line 29, in <module>
    from lsprotocol.types import (
ImportError: cannot import name 'WorkspaceConfigurationParams' from 'lsprotocol.types' (/home/mion00/.pyenv/versions/3.10.8-debug/envs/appdaemon/lib/python3.10/site-packages/lsprotocol/types.py)

	at ChildProcess.exithandler (node:child_process:408:12)
	at ChildProcess.emit (node:events:526:28)
	at maybeClose (node:internal/child_process:1092:16)
	at ChildProcess._handle.onexit (node:internal/child_process:302:5)

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/mion00/.pyenv/versions/3.10.8-debug/envs/appdaemon/lib/python3.10/site-packages/esbonio/lsp/__init__.py", line 52, in <module>
    from .rst import CompletionContext
  File "/home/mion00/.pyenv/versions/3.10.8-debug/envs/appdaemon/lib/python3.10/site-packages/esbonio/lsp/rst/__init__.py", line 37, in <module>
    from pygls.server import LanguageServer
  File "/home/mion00/.pyenv/versions/3.10.8-debug/envs/appdaemon/lib/python3.10/site-packages/pygls/server.py", line 29, in <module>
    from lsprotocol.types import (
ImportError: cannot import name 'WorkspaceConfigurationParams' from 'lsprotocol.types' (/home/mion00/.pyenv/versions/3.10.8-debug/envs/appdaemon/lib/python3.10/site-packages/lsprotocol/types.py)
```

**Comment by mutfuzz** [comment]
I had the same issue, and it was because of the latest release of the python package lsprotocol ([2023.0.0a0](https://github.com/microsoft/lsprotocol/releases/tag/2023.0.0a0)

to solve it:
`pip uninstall lsprotocol`
`pip install lsprotocol==2022.0.0a10`

**Comment by alcarney** [comment]
Some additional context around the issue is available from [this comment](https://github.com/openlawlibrary/pygls/issues/317#issuecomment-1452962422) onwards, if you are interested.

`lsprotocol` has released [v2023.0.0a1](https://github.com/microsoft/lsprotocol/releases/tag/2023.0.0a1) which appears to provide a fix.

**Comment by mion00** [comment]
Thanks for the quick replies! A quick `pip install --upgrade lsprotocol` with the updated version seems to have fixed the issue.
